### PR TITLE
Fix AutoFollowIT.testPauseAndResumeWithMultipleAutoFollowPatterns

### LINKED ISSUE #47917: AutoFollowIT failures
5 tests within AutoFollowIT have been failed today: 

https://gradle-enterprise.elastic.co/s/tgnjxczdo7ave/tests/htwk6wzdfugzg-7uzlsljkhp3ee
https://gradle-enterprise.elastic.co/s/tgnjxczdo7ave/tests/htwk6wzdfugzg-lunr4t27llkji
https://gradle-enterprise.elastic.co/s/jzn6aslxsi4v6/tests/htwk6wzdfugzg-c24zvdrkqeiy2
https://gradle-enterprise.elastic.co/s/jzn6aslxsi4v6/tests/htwk6wzdfugzg-7uzlsljkhp3ee
https://gradle-enterprise.elastic.co/s/hvimqd7ekwr7e/tests/htwk6wzdfugzg-7uzlsljkhp3ee

### PR DESCRIPTION
**Pull request by tlrx**
The test `testPauseAndResumeWithMultipleAutoFollowPatterns` failed multiple times, mostly because it creates too many leader indices and the following cluster cannot cope with cluster state updates generated by following indices creation and pause/resume auto-followers changes.

This PR simplifies the test by creating at most 20 leader indices and by waiting for any new leader index to be picked up by the auto-follower before created another leader index. It also pause and resume less auto-followers as previously.

closes #47917

### CODE REVIEW

**Comment by tlrx** [comment]
@elasticmachine test this please

File: x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java [hunk 1]
```diff
@@ -471,81 +470,86 @@ public void testPauseAndResumeWithMultipleAutoFollowPatterns() throws Exception
...
         final AtomicBoolean running = new AtomicBoolean(true);
-        final Set<String> leaderIndices = ConcurrentCollections.newConcurrentSet();
+        final AtomicInteger leaderIndices = new AtomicInteger(0);
+
+        // start creating new indices on the remote cluster
         final Thread createNewLeaderIndicesThread = new Thread(() -> {
-            while (running.get()) {
+            while (running.get() && leaderIndices.get() < 20) {
+                final String prefix = randomFrom(prefixes);
+                final String leaderIndex = prefix + randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
```
Comment (dnhatn) [inline x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java]: I think we can have duplicate indices here.

**Review by dnhatn — APPROVED** [review]
I left a smaller comment but LGTM. Thanks Tanguy.

**Comment by tlrx** [comment]
@elasticmachine update branch

(Lucene versions tests failed)

File: x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java (hunk 1 above)
Comment (tlrx) [reply x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java]: Right, thanks! I adapted the logic a bit to not create duplicate indices here.

**Comment by tlrx** [comment]
Thanks @dnhatn !

**Pull request merged**